lte/gateway/c/core/oai/tasks/mme_app/mme_app_ip_imsi.cpp

```diff
@@ -15,50 +15,48 @@ limitations under the License.
 #include <unordered_map>
 #include <iostream>
 #include "mme_app_ip_imsi.h"
-using namespace std;
-typedef unordered_map<string, vector<uint64_t>> Ipv4Map;
-
-Ipv4Map ipv4map;
-
-void initialize_ipv4_map() {
-  OAILOG_FUNC_IN(LOG_MME_APP);
-  ipv4map = Ipv4Map{};
-  OAILOG_FUNC_OUT(LOG_MME_APP);
-}
+#include "mme_app_ueip_imsi_map.h"
+#include "mme_app_state_manager.h"
 
+// Description: Logs the content of ueip_imsi map
 void mme_app_log_ipv4_imsi_map() {
   OAILOG_FUNC_IN(LOG_MME_APP);
-  for (auto itr = ipv4map.begin(); itr != ipv4map.end(); ++itr) {
-    for (auto it_vec = itr->second.begin(); it_vec != itr->second.end();
-         ++it_vec) {
+  UeIpImsiMap& ueip_imsi_map =
+      magma::lte::MmeNasStateManager::getInstance().get_mme_ueip_imsi_map();
+  for (const auto& itr_map : ueip_imsi_map) {
+    for (const auto& it_vec : itr_map.second) {
       OAILOG_TRACE(
-          LOG_MME_APP, "ue_ip: %s \t imsi:%lu \n", itr->first.c_str(), *it_vec);
+          LOG_MME_APP, "ue_ip: %s \t imsi:%lu \n", itr_map.first.c_str(),
+          it_vec);
     }
     OAILOG_TRACE(LOG_MME_APP, "\n");
   }
   OAILOG_FUNC_OUT(LOG_MME_APP);
 }
 
 /* Description: ue_ip address is allocated by either roaming PGWs or mobilityd
- * So there is possibility to allocate same ue ip address for the UEs.
+ * So there is possibility to allocate same ue ip address for different UEs.
  * So defining ue_ip_imsi map with key as ue_ip and value as list of imsis
  * having same ue_ip
  */
 int mme_app_insert_ue_ipv4_addr(uint32_t ipv4_addr, imsi64_t imsi64) {
   OAILOG_FUNC_IN(LOG_MME_APP);
+  UeIpImsiMap& ueip_imsi_map =
+      magma::lte::MmeNasStateManager::getInstance().get_mme_ueip_imsi_map();
```

Comment:
minor nit: add `using magma::lte::MmeNasStateManager` as this class is used multiple times on this file

Reply:
Incorporated minor comment